### PR TITLE
[#63 #66 #67 #70] Enhancement / Apply improvements

### LINKED ISSUE #63: Move QueryEngine classes to their own file

### LINKED ISSUE #66: Improve log message
```
 raise RetryException(
hyperon_das.exceptions.RetryException: ('The number of attempts has been exceeded or a timeout has occurred', {'attempts': 5, 'time': '100 seconds'})
```

The name of this exception is terrible and the message itself is not particularly useful. The name is misleading because it's saying that there's a problem related to `retry` but that's not the case. The real problem is that it's not being able to connect. So it should be something like `ConnectionError` or something alike.

As for the information, we should provide here every piece of useful information from the context. The obvious ones being host and port number that were tried. There is probably more to add so please take a careful look at the code and think about what would we want to be there if we only have this exception to debug a reported error.

### LINKED ISSUE #67: Add `normal` log message when trying to connect a remote DAS
I just received a report that a user is not being able to connect to a remote DAS. The first question I asked: "What does the log say?" and he answered: "There's no message in the log at all!"

In this case there should have a log message saying that the user is trying to connect to remote DAS in host X por Y etc etc etc. And then log messages counting retries. And finally another log message saying that the connection definitely failed and providing all information we may find useful to help in debug.

### LINKED ISSUE #70: Check example codes in DAS API methods to make sure they work and are compliant with current API
At least the comment for `get_link()` is wrong as the example is passing node names instead of handles as parameters for targets in get_link()

### PR DESCRIPTION
**Pull request by marcocapozzoli**
- Remove unused Enums
- Remove unused exceptions
- Move engines to a new file
- Return exception in `get_atom`, `get_node` and `get_link`
- Adjustments to API documentation
- Add log to Functions remote connection

Resolves #63 
Resolves #66 
Resolves #67
Resolves #70

### CODE REVIEW

File: hyperon_das/das.py [hunk 1]
```diff
@@ -42,18 +30,16 @@ def __init__(self, **kwargs: Optional[Dict[str, Any]]) -> None:
 
         if query_engine_parameter == 'local':
             self.query_engine = LocalQueryEngine(self.backend, kwargs)
+            logger().info('Initialized local Das')
         elif query_engine_parameter == "remote":
             self.query_engine = RemoteQueryEngine(self.backend, kwargs)
+            logger().info('Initialized remote Das')
```
Comment (andre-senna) [inline hyperon_das/das.py:36]: If you put log message only after constructor you will have nothing in the log if the connection itself breaks. I'm OK having this messages here but I saw no log message before the connection is attempted in RemoteQueryEngine. Please make sure that we have proper logging messages telling that a connection is being attempted either here or in the query engine's constructor.

File: hyperon_das/engines.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
+import json
```
Comment (andre-senna) [inline hyperon_das/engines.py:1]: `engines.py` is not a good name. It's misleading as it gives the impression that any "engines" would be defined in this file, which is not true. `query_engines.py` is a better name IMO.

**Review by andre-senna — CHANGES_REQUESTED** [review]

File: hyperon_das/engines.py (hunk 2 above)
Comment (marcocapozzoli) [reply hyperon_das/engines.py:1]: @andre-senna, but what about our new engine, TraserveEngine? Don't you think this class and its concrete implementations couldn't stay in this file?

File: hyperon_das/das.py (hunk 1 above)
Comment (marcocapozzoli) [reply hyperon_das/das.py:36]: @andre-senna look the decorators.py file, in the `retry` function. I put some logs there and also in the `engines.py file on line 206`. notice that I put some logs like `debug`, do you agree with that?

So, the logs in the file can some like this:

[das.log](https://github.com/singnet/das-query-engine/files/13704355/das.log)

Disclaimer: the line with `Starting new HTTP connection`... it's a log of the `request` lib

File: hyperon_das/engines.py (hunk 2 above)
Comment (marcocapozzoli) [reply hyperon_das/engines.py:1]: fixed

File: hyperon_das/das.py (hunk 1 above)
Comment (andre-senna) [reply hyperon_das/das.py:36]: OK

**Review by andre-senna — APPROVED** [review]

**Pull request merged**